test/defi/sandRewardPool/calculators/landOwnerContributionCalculator.test.ts

```diff
@@ -0,0 +1,76 @@
+import {expect} from '../../../chai-setup';
+import {landOwnerContributionCalculatorSetup} from '../fixtures/contributionCalculator.fixture';
+import {deployments, ethers} from 'hardhat';
+
+describe('LandOwnerContributionCalculator', function () {
+  describe('roles', function () {
+    it('admin should be able to call setNFTMultiplierToken', async function () {
+      const {
+        deployer,
+        contractAsAdmin,
+      } = await landOwnerContributionCalculatorSetup();
+      await deployments.deploy('someOtherContract', {
+        from: deployer,
+        contract: 'ERC721Mintable',
+        args: ['LandToken', 'LTK'],
+      });
+      const someOtherContract = await ethers.getContract('someOtherContract');
+      await expect(
+        contractAsAdmin.setNFTMultiplierToken(someOtherContract.address)
+      ).not.to.be.reverted;
+    });
+    it('others should fail to call setNFTMultiplierToken', async function () {
+      const {
+        deployer,
+        contract,
+        contractAsOther,
+      } = await landOwnerContributionCalculatorSetup();
+      await deployments.deploy('someOtherContract', {
+        from: deployer,
+        contract: 'ERC721Mintable',
+        args: ['LandToken', 'LTK'],
+      });
+      const someOtherContract = await ethers.getContract('someOtherContract');
+      await expect(
+        contract.setNFTMultiplierToken(someOtherContract.address)
+      ).to.be.revertedWith('not the owner');
+      await expect(
+        contractAsOther.setNFTMultiplierToken(someOtherContract.address)
+      ).to.be.revertedWith('not the owner');
+    });
+  });
+
+  describe('calculation', function () {
+    it('users without lands cannot stake', async function () {
+      const {
+        contract,
+        landToken,
+        other,
+      } = await landOwnerContributionCalculatorSetup();
+      expect(await landToken.balanceOf(other)).to.be.equal(0);
+      expect(await contract.multiplierOf(other)).to.be.equal(0);
+      await expect(contract.computeContribution(other, 1000)).to.revertedWith(
+        'no lands'
```

Comment:
We won't be able to update the contribution of a user that get rid of his land

Reply:
you are right, we fixed that.